receiver/promtailreceiver/go.mod

```diff
@@ -1,248 +0,0 @@
-module github.com/open-telemetry/opentelemetry-collector-contrib/receiver/promtailreceiver
```

Comment:
I can create a separate PR with deprecation. And we will merge it, and them we can merge this one

Reply:
Created PR to deprecate promtail receiver first https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18534 to follow the right flow. Let's merge it first and then let's merge this one

Reply:
Thanks @mar4uk! :)

Reply:
This receiver was never released in the contrib distro, so I'm not sure what the right thing to do here. If the wish is to tag a the promtailreceiver so that it shows up in the docs, maybe we should consider doing a patch release of contrib (or of this component alone?) to not block updating dependencies for all other components until the next release.

Reply:
I would prefer to not tag the promtail receiver. If it is tagged community could try to use it and face problems with dependencies.
Yeah, this receiver was never released, that's why I doubted if it needs deprecation instead of just removal...

Reply:
> If the wish is to tag a the promtailreceiver so that it shows up in the docs, maybe we should consider doing a patch release of contrib (or of this component alone?) to not block updating dependencies for all other components until the next release.

Doing a patch release of this component works for me, we also have that problem for `tracegen` as mentioned above, so I guess we can ignore this problem and create an issue to discuss what to do for both (maybe there are even more components with this problem?)

Reply:
I think that we should ensure our public docs are up-to-date.  Since https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/promtailreceiver exists and doesn't say it is deprecated/removed, we should update it to be accurate.  Same with https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/tracegen.

I don't have an opinion on whether that needs to be immediate via a patch or if it can wait for 0.72.0.

Reply:
I've published a tag and go docs now show the package as deprecated. https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18605

@mar4uk if you can rebase we should be good to move ahead w/ the PR!